### PR TITLE
Fix collada textures not being loaded over websocket

### PR DESCRIPTION
**Pull request by achim-k**
**User-Facing Changes**
Fix collada textures not being loaded over websocket

**Description**
Studio loads collada meshes through the asset management layer which, in case of a foxglove-websocket connection, may fetch the asset over the websocket connection. However, this was not the case for texture assets that are referenced by the collada mesh, resulting in the entire collada mesh not being loaded.

This PR fixes this by also loading the referenced texture assets through Studio's asset management layer.

Resolves FG-5340

### CODE REVIEW

File: packages/studio-base/src/panels/ThreeDeeRender/ModelCache.ts [hunk 1]
```diff
@@ -203,8 +203,30 @@ export class ModelCache {
...
+      if (!node.textContent) {
+        continue;
+      }
+
+      try {
+        const textureUrl = new URL(node.textContent, baseUrl(url)).toString();
+        const textureAsset = await this.#fetchAsset(textureUrl);
+        const objectUrl = URL.createObjectURL(
+          new Blob([textureAsset.data], { type: textureAsset.mediaType }),
+        );
```
Comment (achim-k) [inline packages/studio-base/src/panels/ThreeDeeRender/ModelCache.ts:220]: There is currently no way to call [URL.revokeObjectURL()](https://developer.mozilla.org/en-US/docs/Web/API/URL/revokeObjectURL_static) on this...

Comment (jtbandes) [reply packages/studio-base/src/panels/ThreeDeeRender/ModelCache.ts:220]: It wouldn't work to do it after the `itemEnd`?

**Review by jtbandes — APPROVED** [review]

File: packages/studio-base/src/panels/ThreeDeeRender/ModelCache.ts (hunk 1 above)
Comment (achim-k) [reply packages/studio-base/src/panels/ThreeDeeRender/ModelCache.ts:220]: Oh yes, that actually works :+1:

File: packages/studio-base/src/panels/ThreeDeeRender/ModelCache.ts [hunk 2]
```diff
@@ -203,14 +203,40 @@ export class ModelCache {
...
+        textureUrls.set(textureUrl, objectUrl);
+      } catch (e) {
+        log.error(e);
+        onError(node.textContent);
+      }
+    }
+
     const manager = new THREE.LoadingManager(undefined, undefined, onError);
-    manager.setURLModifier(rewriteUrl);
+    manager.setURLModifier((u) => textureUrls.get(u) ?? rewriteUrl(u));
```
Comment (defunctzombie) [inline packages/studio-base/src/panels/ThreeDeeRender/ModelCache.ts:229]: small stylistic nit: I dislike the use of single letter variable names (unless they are math constants). Even in limited settings such as this function definition having a few more characters helps reading by removing ambiguity on what _u_ is suppose to be.

**Pull request merged**